src/ILCompiler.Compiler/src/Compiler/ExportedMethodsRootProvider.cs

```diff
@@ -32,15 +59,15 @@ public void AddCompilationRoots(IRootingServiceProvider rootProvider)
                         if (runtimeExportName != null)
                             rootProvider.AddCompilationRoot(method, "Runtime export", runtimeExportName);
                     }
-
-                    if (ecmaMethod.IsNativeCallable)
-                    {
-                        string nativeCallableExportName = ecmaMethod.GetNativeCallableExportName();
-                        if (nativeCallableExportName != null)
-                            rootProvider.AddCompilationRoot(method, "Native callable", nativeCallableExportName);
-                    }
                 }
             }
+
+            foreach (var ecmaMethod in ExportedMethods)
```

Comment:
I do not think we need to be differentiating between NativeCallable and RuntimeExports here. They can be treated the same.